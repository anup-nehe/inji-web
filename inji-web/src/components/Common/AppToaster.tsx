import React from "react";
import { ToastContainer, toast } from "react-toastify";
import { useSelector } from "react-redux";
import { RootState } from "../../types/redux";
import { isRTL } from "../../utils/i18n";

export const AppToaster: React.FC = () => {
    const language = useSelector((state: RootState) => state.common.language);
    const positionClass = isRTL(language) ? "top-left" : "top-right";

    React.useEffect(() => {
        toast("AppToaster test message");
    }, []);

    return (
        <div className={`text-center pt-20 pb-10 ${positionClass}`} data-testid="Apptoaster-outer-container">
            <ToastContainer
                position={isRTL(language) ? "top-left" : "top-right"}
                autoClose={1000}
                hideProgressBar
                newestOnTop
                closeOnClick
                rtl={isRTL(language)}
                icon={<React.Fragment />}
                pauseOnFocusLoss
                draggable
                pauseOnHover
                style={{ width: '400px' }}
                theme="colored"
            />
        </div>
    );
};
