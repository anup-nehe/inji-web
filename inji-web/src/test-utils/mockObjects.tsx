import { IssuerWellknownObject } from '../types/data';
import { LandingPageWrapperProps } from '../components/Common/LandingPageWrapper';
import { NavBarProps } from '../types/components';
export const mockDisplayArrayObject = {
    name: "Name",
    language: "en",
    locale: "en",
    logo: {
        url: "https://url.com",
        alt_text: "alt text of the url"
    },
    title: "Title",
    description: "Description",
};

export const mockCredentials: IssuerWellknownObject = {
    credential_issuer: "",
    credential_endpoint: "",
    authorization_servers: [""],
    credential_configurations_supported: {
        InsuranceCredential: {
            format: "ldp_vc",
            scope: "mosip_ldp_vc",
            order: [],
            display: [{
                name: "Name1",
                language: "en",
                locale: "en",
                logo: {
                    url: "https://url.com",
                    alt_text: "alt text of the url"
                },
                title: "Title",
                description: "Description",
            }],
            proof_types_supported: [],
            credential_definition: {
                type: [],
                credentialSubject: {
                    fullName: {
                        display: [{
                            name: "Name1",
                            language: "en",
                            locale: "en",
                            logo: {
                                url: "https://url.com",
                                alt_text: "alt text of the url"
                            },
                            title: "Title",
                            description: "Description",
                        }],
                    }
                }
            }
        },
        AnotherCredential: {
            format: "ldp_vc",
            scope: "mosip_ldp_vc",
            order: [],
            display: [{
                name: "Name2",
                language: "en",
                locale: "en",
                logo: {
                    url: "https://url.com",
                    alt_text: "alt text of the url"
                },
                title: "Title",
                description: "Description",
            }],
            proof_types_supported: [],
            credential_definition: {
                type: [],
                credentialSubject: {
                    fullName: {
                        display: [{
                            name: "Name2",
                            language: "en",
                            locale: "en",
                            logo: {
                                url: "https://url.com",
                                alt_text: "alt text of the url"
                            },
                            title: "Title",
                            description: "Description",
                        }],
                    }
                }
            }
        }
    }
};

export const mockSearchCredential ={
    credential_configurations_supported: {
        InsuranceCredential: {
            display: [{
                name: "Insurance Credential",
                language: "en",
                locale: "en",
                logo: {
                    url: "https://url.com",
                    alt_text: "alt text of the url"
                },
                title: "Title",
                description: "Description",
            }],
        },
        AnotherCredential: {
            display: [{
                name: "Another Credential",
                language: "en",
                locale: "en",
                logo: {
                    url: "https://url.com",
                    alt_text: "alt text of the url"
                },
                title: "Title",
                description: "Description",
            }],
        }
    }
};

export const mockIssuerObject = {
        name: 'Test Issuer',
        desc: 'Test Description',
        protocol: 'OTP' as 'OTP', // Explicitly set the type to 'OTP' or 'OpenId4VCI'
        issuer_id: 'test-issuer',
        authorization_endpoint: 'https://auth.test.com',
        credentials_endpoint: 'https://credentials.test.com',
        display: [{
            name: "Name",
            language: "en",
            locale: "en",
            logo: {
                url: "https://url.com",
                alt_text: "alt text of the url"
            },
            title: "Title",
            description: "Description",
        }],
        client_id: 'test-client-id',
        redirect_uri:'test-redirect-uri',
        token_endpoint:'test-token_endpoint',
        proxy_token_endpoint:'test-proxy_token_endpoint',
        client_alias:'',
        ovp_qr_enabled: true,
        scopes_supported: ['scope1', 'scope2']
}

export const mockIssuerObjectList =  [
    {
        name: 'Issuer 1',
        desc: 'Description 1',
        protocol: 'OpenId4VCI',
        issuer_id: 'issuer1',
        authorization_endpoint: 'https://issuer1.com/auth',
        credentials_endpoint: 'https://issuer1.com/credentials',
        display: [
            {
                name: 'Issuer 1',
                language: 'en',
                locale: 'en-US',
                logo: { url: 'https://issuer1.com/logo.png', alt_text: 'Issuer 1 Logo' },
                title: 'Issuer 1 Title',
                description: 'Issuer 1 Description',
            },
        ],
        client_id: 'client1',
        redirect_uri: 'https://issuer1.com/redirect',
        token_endpoint: 'https://issuer1.com/token',
        proxy_token_endpoint: 'https://issuer1.com/proxy-token',
        client_alias: 'client1-alias',
        ovp_qr_enabled: true,
        scopes_supported: ['openid', 'profile'],
    },
    {
        name: 'Issuer 2',
        desc: 'Description 2',
        protocol: 'OpenId4VCI',
        issuer_id: 'issuer2',
        authorization_endpoint: 'https://issuer2.com/auth',
        credentials_endpoint: 'https://issuer2.com/credentials',
        display: [
            {
                name: 'Issuer 2',
                language: 'en',
                locale: 'en-US',
                logo: { url: 'https://issuer2.com/logo.png', alt_text: 'Issuer 2 Logo' },
                title: 'Issuer 2 Title',
                description: 'Issuer 2 Description',
            },
        ],
        client_id: 'client2',
        redirect_uri: 'https://issuer2.com/redirect',
        token_endpoint: 'https://issuer2.com/token',
        proxy_token_endpoint: 'https://issuer2.com/proxy-token',
        client_alias: 'client2-alias',
        ovp_qr_enabled: false,
        scopes_supported: ['openid', 'profile'],
    },
]

export const mockLandingPageWrapperProps: LandingPageWrapperProps = {
    icon: <div data-testid="Test-Icon">Icon</div>,
    title: "Test Title",
    subTitle: "Test SubTitle",
    gotoHome: true,
};

export const mockNavBarProps: NavBarProps = {
    title: 'Test Title',
    link: '/test-link',
    search: true,
};