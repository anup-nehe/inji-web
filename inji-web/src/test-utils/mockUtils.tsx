import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { reduxStore } from '../redux/reduxStore';

export const mockUseTranslation = () => {
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
            t: (key: string) => key,
        }),
    }));
};

export const mockUseNavigate = () => {
    const mockNavigate = jest.fn();
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useNavigate: mockNavigate,
    }));
};

export const mockUseSearchCredentials = () => {
    jest.mock('../components/Credentials/SearchCredential', () => ({
        SearchCredential: () => <></>,
    }));
};

export const mockUseSelector = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useSelector: jest.fn(),
    }));
};

export const setMockUseSelectorState = (state: any) => {
    const useSelectorMock = require('react-redux').useSelector;
    useSelectorMock.mockImplementation((selector: any) => selector(state));
};

export const mockUseGetObjectForCurrentLanguage = () => {
    jest.mock('../utils/i18n', () => ({
        getObjectForCurrentLanguage: jest.fn(),
    }));
};

export const wrapUnderRouter = (children: React.ReactNode) => {
    return <Router>{children}</Router>;
};

export const mockUseToast = () => {
    jest.mock('react-toastify', () => ({
        toast: jest.fn(),
        ToastContainer: jest.requireActual('react-toastify').ToastContainer,
    }));
};

export const mockUseDispatch = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useDispatch: jest.fn(),
    }));
};

export const mockUseSpinningLoader = () => {
    jest.mock('../components/Common/SpinningLoader', () => ({
        SpinningLoader: () => <></>,
    }));
};

export const mockUseLanguageSelector = () => {
    jest.mock('../components/Common/LanguageSelector', () => ({
        LanguageSelector: () => <></>,
    }));
};

interface RenderWithProviderOptions extends Omit<RenderOptions, 'queries'> {}

export const renderWithProvider = (element: ReactElement, options?: RenderWithProviderOptions) => {
    return render(
        <Provider store={reduxStore}>
            <Router>
                {element}
            </Router>
        </Provider>,
        options
    );
};

