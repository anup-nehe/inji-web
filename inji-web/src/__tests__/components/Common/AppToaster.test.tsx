import React from 'react';
import { AppToaster } from '../../../components/Common/AppToaster';
import { reduxStore } from '../../../redux/reduxStore'; // Assuming the store is named reduxStore
import { renderWithProvider } from '../../../test-utils/mockUtils';
import { toast } from 'react-toastify';

// todo : extract the local method to mockUtils,where added because of mock is not being properly mocked which leads to match error
jest.mock('react-toastify', () => ({
    toast: jest.fn(),
    ToastContainer: jest.requireActual('react-toastify').ToastContainer,
}));

describe('Testing the Layout of AppToaster', () => {
    test('Check if the layout is matching with the snapshots for English language', () => {
        const { asFragment } = renderWithProvider(<AppToaster />);
        reduxStore.dispatch({ type: 'SET_LANGUAGE', payload: 'en' });
        expect(asFragment()).toMatchSnapshot();
    });

    test('Check if the layout is matching with the snapshots for Arabic language', () => {
        const { asFragment } = renderWithProvider(<AppToaster />);
        reduxStore.dispatch({ type: 'SET_LANGUAGE', payload: 'ar' });
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('Testing the Functionality of AppToaster', () => {
    test('Check if toast is called with the correct message for English language', () => {
        renderWithProvider(<AppToaster />);
        reduxStore.dispatch({ type: 'SET_LANGUAGE', payload: 'en' });
        expect(toast).toHaveBeenCalledWith("AppToaster test message");
    });

    test('Check if toast is called with the correct message for Arabic language', () => {
        renderWithProvider(<AppToaster />);
        reduxStore.dispatch({ type: 'SET_LANGUAGE', payload: 'ar' });
        expect(toast).toHaveBeenCalledWith("AppToaster test message");
    });

    afterEach(()=>{
        jest.clearAllMocks();
    })
});
