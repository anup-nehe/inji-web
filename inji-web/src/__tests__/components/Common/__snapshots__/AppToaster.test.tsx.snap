// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Layout of AppToaster Check if the layout is matching with the snapshots for Arabic language 1`] = `
<DocumentFragment>
  <div
    class="text-center pt-20 pb-10 top-right"
    data-testid="Apptoaster-outer-container"
  >
    <div
      class="Toastify"
    />
  </div>
</DocumentFragment>
`;

exports[`Testing the Layout of AppToaster Check if the layout is matching with the snapshots for English language 1`] = `
<DocumentFragment>
  <div
    class="text-center pt-20 pb-10 top-right"
    data-testid="Apptoaster-outer-container"
  >
    <div
      class="Toastify"
    />
  </div>
</DocumentFragment>
`;
